app/bundles/EmailBundle/Controller/AjaxController.php

```diff
@@ -285,4 +286,56 @@ public function getEmailDeliveredCountAction(Request $request, CacheProvider $ca
             'delivered'   => $deliveredCount,
         ]);
     }
+
+    public function heatmapAction(Request $request): JsonResponse
+    {
+        $emailId = (int) InputHelper::clean($request->query->get('id'));
+
+        /** @var EmailModel $model */
+        $model = $this->getModel('email');
+
+        $email       = $model->getEntity($emailId);
+
+        if (null === $email) {
+            return $this->sendJsonResponse([
+                'success' => 0,
```

Comment:
I'd also like to avoid this `success` thing that Mautic has in many Ajax responses. Developers should use HTTP response code instead to check for success. Let's remove it
```suggestion
```